node/grpc/v2_metrics.go

```diff
@@ -0,0 +1,202 @@
+package grpc
+
+import (
+	"fmt"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	grpcprom "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"google.golang.org/grpc"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"time"
+)
+
+const namespace = "eigenda_node"
+
+// V2Metrics encapsulates metrics for the v2 DA node.
+type V2Metrics struct {
+	logger logging.Logger
+
+	registry         *prometheus.Registry
+	server           *http.Server
+	grpcServerOption grpc.ServerOption
+
+	storeChunksLatency  *prometheus.SummaryVec
+	storeChunksDataSize *prometheus.GaugeVec
+
+	getChunksLatency  *prometheus.SummaryVec
+	getChunksDataSize *prometheus.GaugeVec
+
+	dbSize           *prometheus.GaugeVec
+	dbSizePollPeriod time.Duration
+	dbDir            string
+	isAlive          *atomic.Bool
+}
+
+// NewV2Metrics creates a new V2Metrics instance. dbSizePollPeriod is the period at which the database size is polled.
+// If set to 0, the database size is not polled.
+func NewV2Metrics(
```

Comment:
Is this compatible with v1 metrics? 
v1 metrics also registers mux and initializes its own server

Reply:
Ok, I've now made it so we use the same registry/server as v1. The result is a bit hacky since the v1 metrics code is in another repo.